packages/dds/map/src/directory.ts

```diff
@@ -1869,15 +1899,19 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
     /**
      * Clear all keys in memory in response to a remote clear, but retain keys we have modified but not yet been ack'd.
      */
-    private clearExceptPendingKeys(local: boolean) {
+    private clearExceptPendingKeys(local: boolean): void {
         // Assuming the pendingKeys is small and the map is large
         // we will get the value for the pendingKeys and clear the map
         const temp = new Map<string, ILocalValue>();
+
+        // eslint-disable-next-line unicorn/no-array-for-each
         this.pendingKeys.forEach((value, key, map) => {
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
             temp.set(key, this._storage.get(key)!);
         });
         this.clearCore(local);
+
+        // eslint-disable-next-line unicorn/no-array-for-each
         temp.forEach((value, key, map) => {
             this.setCore(key, value, true);
         });
```

Comment:
`map` from the callback is not being used, so we could replace these with `for [key, value] of ...`, right?